### DISCUSSION TITLE
WSL2 with Docker Desktop support

### DISCUSSION THREAD

**Discussion by roughnecks** [henrygd/beszel#76] [General]
Hi,
I have a Debian WSL2 install on Windows 10 and I also installed Docker Desktop with WSL2 support.
Running `beszel-agent` in a terminal works fine using an external HUB (like on a VPS); the only requirement is to open the port on your router (port-forwarding), on Windows Firewall and last, but not least, on WSL itself:

Open a CMD prompt with admin permissions and run:
`netsh interface portproxy set v4tov4 listenport=45876 listenaddress=0.0.0.0 connectport=45876 connectaddress=<wsl2 IP>`

to get `<wsl2 IP>` run:
`ip a` and look for `inet' on eth0

`inet 172.27.22.69/20 brd 172.27.31.255 scope global eth0`
"172.27.22.69" in my case

If you have multiple VMs, chose the right one, obviously.

As for the filesystem, I ran:
`df`

and looked for "mounted on /"

Just to be clear, containers running in Docker Desktop are successfully shown in beszel stats.
I haven't tried running the agent in Docker, because it looked like over-complicating things.

Cheers

**Comment by henrygd** [comment]
Great, thanks for the info! Does this give you the correct CPU usage and memory of the windows host as well?

**Comment by roughnecks** [comment]
Well, CPU name is reported correctly, so I assume yes, while for memory it says "total" 8GB, while I have 16 on the host. If I run `free -h` on WSL2 though..

```
free -h
               total        used        free      shared  buff/cache   available
Mem:           7.7Gi       1.4Gi       6.0Gi       101Mi       636Mi       6.3Gi
Swap:          2.0Gi          0B       2.0Gi
```

Not sure who/what allocated that mem to the VM, I don't remember having set anything up myself and this Debian guest was originally WSL1.

**Comment by roughnecks** [comment]
![image](https://github.com/user-attachments/assets/bd9abb65-d54c-4e5a-93c6-48b331c570a6)

One thing that could be improved is disk space. It grabs "/" but in `df -h` it reports 1TB and so in Beszel, but I have a 320GB SSD where Windows and WSL run and a 2TB Data HDD, so I'm really puzzled here.

`/dev/sdc                                 1007G   18G  939G   2% /`

**Comment by roughnecks** [comment]
Also btop doesn't get disk info:

![image](https://github.com/user-attachments/assets/9ab661e1-77cf-42e9-a682-2d13e9f5bbfb)
